### PR TITLE
=core Remove deperecated usages of CompositeFuture.

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:

Remove the deprecated usages.

Result:
Clean code.

### CODE REVIEW

**Review by tsegismont — COMMENTED** [review]
Thank you @He-Pin

File: src/main/java/io/vertx/core/impl/DeploymentManager.java [hunk 1]
```diff
@@ -326,13 +325,13 @@ public synchronized Future<Void> doUndeploy(ContextInternal undeployingContext)
         return doUndeployChildren(undeployingContext).compose(v -> doUndeploy(undeployingContext));
       } else {
         status = ST_UNDEPLOYED;
-        List<Future> undeployFutures = new ArrayList<>();
+        final List<Future<?>> undeployFutures = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (He-Pin) [reply src/main/java/io/vertx/core/impl/DeploymentManager.java:328]: I see

[the remark this answers is not in the document]
Comment (He-Pin) [reply src/main/java/io/vertx/core/impl/DeploymentManager.java:328]: i will update it tomorrow.

**Comment by He-Pin** [comment]
@tsegismont I have updated this PR, would you like to review this?

**Pull request merged**